docs/gpg.md

```diff
@@ -0,0 +1,29 @@
+# GPG Keys
+
+## Summary
+
+After our security incident in July 2014, we planned to try and contain the scope of our GPG keys to avoid resigning lots of content if (or rather, when) a key is compromised or has to be revoked.
+
+* General use time based keys: for use with Debian archives, nightly packages etc.  Cycled every two years.
+* Major release (1.5, 1.6 etc) based keys: for use with tarballs, RPMs.  Expiry of one year.
+
+## Generating a new key
+
+`generate_gpg` from [theforeman-rel-eng](https://github.com/theforeman/theforeman-rel-eng/) can be used to generate new keys.
+
+See [Generating a new GPG Key for a X.Y release](https://github.com/theforeman/theforeman-rel-eng/#generating-a-new-gpg-key-for-a-xy-release) and [Generating a new GPG Key for signing the Debian repository](https://github.com/theforeman/theforeman-rel-eng/#generating-a-new-gpg-key-for-signing-the-debian-repository) for documentation how to do so.
+
+## Distributing keys
+
+### Release based keys
+
+RPM users are told in install & upgrade documentation to install foreman-release from the new release, which can contain the keys for that release, making distribution easy.
+
+### Time based keys
```

Comment:
Is this done via Puppet or still manual? If Puppet, can we link to a similar commit?

Reply:
"this" being adjusting `freight.conf`? 

7680053bb8fa5f682a41d912ccd89701d0f5b93f - Add 2016 archive key
9f50f62e0fbb120e5c1ce5b36d1d14e1aefb4949 - Remove 2014 archive signing GPG key

Reply:
Basically all steps in this chapter. So also making sure `foreman.asc` is correct on https://deb.theforeman.org/. Ideally that after you've read this, you know how to perform the procedure.